lib/ansible/cli/doc.py

```diff
@@ -63,7 +65,196 @@ class PluginNotFound(Exception):
     pass
 
 
-class DocCLI(CLI):
+class RoleMixin(object):
+    """A mixin containing all methods relevant to role argument specification functionality.
+
+    Note: The methods for actual display of role data are not present here.
+    """
+
+    ROLE_ARGSPEC_FILE = 'argument_specs.yml'
+
+    def _load_argspec(self, role_name, collection_path=None, role_path=None):
+        if collection_path:
+            path = os.path.join(collection_path, 'roles', role_name, 'meta', self.ROLE_ARGSPEC_FILE)
+        elif role_path:
+            path = os.path.join(role_path, 'meta', self.ROLE_ARGSPEC_FILE)
+        else:
+            raise AnsibleError("A path is required to load argument specs for role '%s'" % role_name)
+
+        loader = DataLoader()
+        return loader.load_from_file(path, cache=False, unsafe=True)
+
+    def _find_all_normal_roles(self, role_paths, name_filters=None):
+        """Find all non-collection roles that have an argument spec file.
+
+        :param role_paths: A tuple of one or more role paths.
+        :param name_filters: A tuple of one or more role names used to filter the results.
+
+        :returns: A set of tuples consisting of: role name, full role path
+        """
+        found = set()
+        for path in role_paths:
+            if not os.path.isdir(path):
```

Comment:
This seems like it would be noisy since the default role paths don't necessarily exist.

Reply:
Good point. I could have swore I copied this bit from code somewhere else, but now I can't find it.

Reply:
Removed in latest commits.